samples/snippets/src/main/java/com/example/bigquery/QueryExternalTableAws.java

```
@@ -0,0 +1,68 @@
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.CsvOptions;
+import com.google.cloud.bigquery.DatasetId;
+import com.google.cloud.bigquery.ExternalTableDefinition;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableInfo;
+import com.google.cloud.bigquery.TableResult;
+
+// Sample to queries an external data source aws s3 using a permanent table
+public class QueryExternalTableAws {
+
+  public static void main(String[] args) {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "MY_PROJECT_ID";
+    String datasetName = "MY_DATASET_NAME";
+    String externalTableName = "MY_EXTERNAL_TABLE_NAME";
+    String query =
+        String.format(
+            "SELECT * FROM s%.%s.%s WHERE name LIKE 'W%%'",
+            projectId, datasetName, externalTableName);
+    queryExternalTableAws(query);
+  }
+
+  public static void queryExternalTableAws(String query) {
+    try {
+      // Initialize client that will be used to send requests. This client only needs to be created
+      // once, and can be reused for multiple requests.
+      BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+
+      // Example query to find states starting with 'W'
+      TableResult results = bigquery.query(QueryJobConfiguration.of(query));
+
+      results
+          .iterateAll()
+          .forEach(row -> row.forEach(val -> System.out.printf("%s,", val.toString())));
+
+      System.out.println("Query on aws external permanent table performed successfully.");
+    } catch (BigQueryException | InterruptedException e) {
+      System.out.println("Query not performed \n" + e.toString());
+    }
```

Should these be caught?  Perhaps just mention them or let them propagate?  NOTE - BigQueryException is a RuntimeException and InterruptedException is not.

---

More important is why might these have been thrown and what should a user really do with them?

---

I think we will catch BigQueryException and not InterruptedException. 
Updated!